main.cpp

```diff
@@ -63,7 +63,9 @@ void kv_store_global_api_example()
     res = kv_reset("/kv/");
     printf("kv_reset -> %d\n", err_code(res));
 
-    if (strcmp(STR(MBED_CONF_STORAGE_STORAGE_TYPE), "TDB_EXTERNAL") == 0) {
+    if (strcmp(STR(MBED_CONF_STORAGE_STORAGE_TYPE), "TDB_INTERNAL") != 0) {
```

Comment:
Was the intention the same originally but with an incorrect condition?

Reply:
The original condition was incomplete. The condition should be "anything except TDB_INTERNAL", which includes both TDB_EXTERNAL and FILESYSTEM.

Reply:
So the intention was the same?

Reply:
Yes

Reply:
> The original condition was incomplete.

How did you complete the condition? It [the condition] is now the opposite of what it was before?

Reply:
It's not the opposite. Notice `TDB_EXTERNAL`, `==` before, and `TDB_INTERNAL`, `!=` now.